.github/workflows/stacks-signer-tests.yml

```diff
@@ -0,0 +1,95 @@
+## Run Stacks Signer Unit Tests
+name: Tests::Stacks Signer Unit Tests
+
+on:
+  workflow_call:
+
+## env vars are transferred to composite action steps
+env:
+  BITCOIND_TEST: 0
+  RUST_BACKTRACE: full
+  SEGMENT_DOWNLOAD_TIMEOUT_MINS: 3
+  TEST_TIMEOUT: 30
+
+concurrency:
+  group: stacks-signer-tests-${{ github.head_ref || github.ref || github.run_id}}
+  ## Only cancel in progress if this is for a PR
+  cancel-in-progress: ${{ github.event_name == 'pull_request' }}
+
+jobs:
+  generate-tests:
+    name: Generate JSON of tests to run
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      ## Setup test environment
+      - name: Setup Test Environment
+        id: setup_tests
+        uses: stacks-network/actions/stacks-core/testenv@main
+        with:
+          btc-version: "25.0"
+      - name: Generate tests JSON
+        id: generate_tests_json
+        # List all of the tests using the nextest archive (so we don't need to do another build task)
+        # Filter them such that we only select tests from `--bin stacks-signer`
+        # Transform the output JSON into something that can be used as the matrix input
+        run: |
```

Comment:
hmm,this results in quite a few unit tests. i wonder if a partitioned test run may be better suited here (similat to stacks-core unit tests)

Reply:
These tests are quick. They run in <10s for me locally, so this seems overkill. I think we just stick with running them all together. The overhead for running them this way is far too high.